x-pack/plugins/enterprise_search/public/applications/app_search/components/result/result.tsx

```diff
@@ -56,34 +55,53 @@ export const Result: React.FC<Props> = ({
     [result]
   );
   const numResults = resultFields.length;
-  const typeForField = (fieldName: string) => {
-    if (schemaForTypeHighlights) return schemaForTypeHighlights[fieldName];
-  };
-
   const documentLink = generateEncodedPath(ENGINE_DOCUMENT_DETAIL_PATH, {
     engineName: resultMeta.engine,
     documentId: resultMeta.id,
   });
-  const conditionallyLinkedArticle = (children: React.ReactNode) => {
-    return shouldLinkToDetailPage ? (
-      <ReactRouterHelper to={documentLink}>
-        <article className="appSearchResult__content appSearchResult__content--link">
-          {children}
-        </article>
-      </ReactRouterHelper>
-    ) : (
-      <article className="appSearchResult__content">{children}</article>
-    );
+
+  const typeForField = (fieldName: string) => {
+    if (schemaForTypeHighlights) return schemaForTypeHighlights[fieldName];
   };
 
-  const classes = classNames('appSearchResult', {
-    'appSearchResult--link': shouldLinkToDetailPage,
-  });
+  const ResultActions = () => {
+    if (!shouldLinkToDetailPage && !actions.length) return null;
+    return (
+      <EuiFlexItem grow={false}>
+        <EuiFlexGroup gutterSize="s">
+          {shouldLinkToDetailPage && (
+            <ReactRouterHelper to={documentLink}>
+              <EuiFlexItem grow={false}>
+                <EuiButtonIcon
+                  iconType="eye"
+                  aria-label={i18n.translate(
+                    'xpack.enterpriseSearch.appSearch.result.documentDetailLink',
+                    { defaultMessage: 'Visit document details' }
+                  )}
+                />
+              </EuiFlexItem>
+            </ReactRouterHelper>
+          )}
+          {actions.map(({ onClick, title, iconType, iconColor }) => (
+            <EuiFlexItem key={title}>
+              <EuiButtonIcon
+                iconType={iconType}
+                onClick={onClick}
+                color={iconColor ? 'primary' : undefined}
```

Comment:
Ran across this while writing tests ... I assume this is a mistake and should be:

```suggestion
                color={iconColor ? iconColor : 'primary'}
```

Can you confirm?

Reply:
Ha. Yes. `iconColor : 'primary'` 🤦🏼